lib/workers/repository/update/branch/index.ts

```diff
@@ -553,26 +553,34 @@ export async function processBranch(
     await setStability(config);
     await setConfidence(config);
 
-    // break if we pushed a new commit because status check are pretty sure pending but maybe not reported yet
+    // new commit means status check are pretty sure pending but maybe not reported yet
+    // if PR has not been created + new commit + prCreation !== immediate skip
     // but do not break when there are artifact errors
     if (
-      !config.artifactErrors?.length &&
-      !userRebaseRequested &&
       commitSha &&
-      config.prCreation !== 'immediate'
+      config.prCreation !== 'immediate' &&
+      !branchPr &&
+      !config.artifactErrors?.length &&
+      !userRebaseRequested
     ) {
       logger.debug(`Branch status pending, current sha: ${commitSha}`);
       return {
         branchExists: true,
         updatesVerified,
-        prNo: branchPr?.number,
         result: BranchResult.Pending,
         commitSha,
       };
     }
 
     // Try to automerge branch and finish if successful, but only if branch already existed before this run
-    if (branchExists || config.ignoreTests) {
+    // skip if we have a non-immediate pr and there is an existing PR,
+    // we want to update the PR and skip the Auto merge since status checks aren't done yet
+    if (
+      !commitSha &&
+      !config.artifactErrors?.length &&
+      !userRebaseRequested &&
+      (branchExists || config.ignoreTests)
+    ) {
```

Comment:
we can automerge when new commit and config.ignoreTests.

branchExists only means branch was there or new created.

Reply:
```suggestion
    // skip if we have a non-immediate pr and there is an existing PR,
    // we want to update the PR and skip the Auto merge since status checks aren't done yet
    if (
      (!commitSha &&
      !config.artifactErrors?.length) ||
      config.ignoreTests
    ) {
```

maybe this is the right condition?

Reply:
what about a flow with Artifact Errors + ignoreTests=true ? is it ok to merge with artifact errors?

Reply:
good catch, we shouldn't.

Reply:
if u look up a bit on the same IF statement before automerge check there's:
` if (config.artifactErrors?.length) {`
so basically we dont need to check for artifact errors in there.

ignoreTests is not needed as a check here neither since it's being checked in the branch status
```
 const branchStatus = await resolveBranchStatus(
    config.branchName!,
    config.ignoreTests
  );
```

!commitSha i'm not sure since it's part of the return
```
  const prAutomergeResult = await checkAutoMerge(pr, config);
          if (prAutomergeResult?.automerged) {
            return {
              branchExists,
              result: BranchResult.Automerged,
              commitSha,
            };
          }
```

so the first branch auto merge condition should be 
`    if (!config.artifactErrors?.length && (!commitSha || (commitSha && config.ignoreTests))) {`

either there's no new commit, or there's a commit with ignoreTests

otherwise the ignoreTests shouldnt even exist in this condition. like :
`    if (!config.artifactErrors?.length && !commitSha) {`

WDYT?